docs/how-to-guide/how-to-guide.md

```diff
@@ -0,0 +1,358 @@
+
+```
+Name:             wordpress-k8s-ingress
+Labels:           app.juju.is/created-by=nginx-ingress-integrator
+Namespace:        wordpress-tutorial
+Address:
+Ingress Class:    <none>
+Default backend:  <default>
+Rules:
+  Host           Path  Backends
+  ----           ----  --------
+  wordpress-k8s
+                 /   wordpress-k8s-service:80
+Annotations:     nginx.ingress.kubernetes.io/enable-modsecurity: true
+                 nginx.ingress.kubernetes.io/enable-owasp-modsecurity-crs: true
+                 nginx.ingress.kubernetes.io/modsecurity-snippet:
+                   SecRuleEngine On
+                   SecAction "id:900130,phase:1,nolog,pass,t:none,setvar:tx.crs_exclusions_wordpress=1"
+
+                   Include /etc/nginx/owasp-modsecurity-crs/nginx-modsecurity.conf
+                 nginx.ingress.kubernetes.io/proxy-body-size: 20m
+                 nginx.ingress.kubernetes.io/rewrite-target: /
+                 nginx.ingress.kubernetes.io/ssl-redirect: false
+Events:          <none>
+```
+
+Note the `nginx.ingress.kubernetes.io/enable-modsecurity: true` annotation.
+
+### Configure hostname
+
+To configure a different hostname for WordPress, you can configure the ingress hostname through the
+wordpress-k8s configuration.
+
+```
+juju config wordpress-k8s blog_hostname=<desired-hostname>
+```
+
+To test locally, append the service IP of nginx-ingress-integrator to `/etc/hosts` file with your
+new hostname and curl the service using the hostname.
+
+The output of `juju status` should look similar to the following:
+
+```
+Model  Controller          Cloud/Region        Version  SLA          Timestamp
+wordpress-k8s-tutorial    microk8s-localhost  microk8s/localhost  2.9.38   unsupported
+
+App                       Version              Status  Scale  Charm     Channel  Rev  Address         Exposed  Message
+mysql-k8s                                      active      1  mysql-k8s stable    35  <mysql-ip>  no       ready
+nginx-ingress-integrator                       active      1  nginx-ingress-integrator  stable    54  <nginx-ingress-integrator-ip>   no       Service IP(s): <nginx-ingress-integrator-service-ip>
+wordpress-k8s                                  active      1  wordpress-k8s 16  <wordpress-k8s-ip>  no
+
+Unit                         Workload  Agent  Address       Ports     Message
+mysql-k8s/0*                active    idle   <mysql-k8s-ip>
+nginx-ingress-integrator/0*  active    idle  <nginx-ingress-integrator-ip>            Service IP(s
+): <nginx-ingress-integrator-service-ip>
+wordpress-k8s/0*             active    idle   <wordpress-k8s-ip>
+```
+
+Note the Service IP(s): next to nginx-ingress-integrator charm’s Status output.
+
+Append the new hostname with the ingress IP with following command.
+
+```
+echo <nginx-ingress-integrator-service-ip> <desired-hostname> | sudo tee -a /etc/hosts
+```
+
+Test the ingress by sending a GET request to the hostname.
+
+```
+curl <desired-hostname>
```

Comment:
If `curl` is used in testing the ingress connectivity, you can skip the updating `/etc/hosts` part and use the following command instead:
```bash
curl -H "Host: <desired-hostname>" http://<nginx-ingress-integrator-service-ip>
```

Reply:
Seems like a more concise option, replaced.
Thank you!